internal/common/resource.go

```diff
@@ -392,6 +398,31 @@ func logOperation(result OperationResult, resource client.Object, logger logr.Lo
 	}
 }
 
+func createOperationMetrics(req *Request, result OperationResult, expected client.Object, found client.Object, logger logr.Logger) {
+	if result == OperationResultNone || isVersionUpgrade(req, result, expected, found) {
+		return
+	}
+
+	message := fmt.Sprintf("Created %s resource: %s", expected.GetObjectKind().GroupVersionKind().Kind, expected.GetName())
+	if result == OperationResultUpdated {
+		message = fmt.Sprintf("Updated %s resource: %s", expected.GetObjectKind().GroupVersionKind().Kind, expected.GetName())
+	} else if result == OperationResultDeleted {
+		message = fmt.Sprintf("Deleted %s resource: %s", found.GetObjectKind().GroupVersionKind().Kind, found.GetName())
+	}
+
+	logger.Info(message)
+	TemplatesRestored.Inc()
+}
+
+func isVersionUpgrade(req *Request, result OperationResult, expected client.Object, found client.Object) bool {
+	isOperatorUpgrade := req.Instance.Status.ObservedVersion != GetOperatorVersion()
+
+	expectedVersion := expected.GetLabels()[AppKubernetesVersionLabel]
+	foundVersion := found.GetLabels()[AppKubernetesVersionLabel]
```

Comment:
At this point in the code the labels on `expected` and `found` are guaranteed to be the same, because the labels on `found` are updated here on line 190.